src/TLuaInterpreter.cpp

```diff
@@ -10641,6 +10652,501 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+			lua_error(L);
+			return 1;
+		}
+		
+		index = lua_tonumber(L, 2);
+		index--;
+		
+		if(index < 0)
+			index = 0;
+		
+		if(index > speechQueue.size())
+			index = speechQueue.size();
+	}
+	else {
+		index = speechQueue.size();
+	}
+
+	speechQueue.insert(index, inputText);
+	
+	TEvent event;
+	event.mArgumentList.append(QLatin1String("ttsSpeechQueued"));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(QLatin1String(inputText));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+	event.mArgumentList.append(QLatin1String(index));
+	event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+	mpHost->raiseEvent(event);
+	
+	TLuaInterpreter::ttsStateChanged(speechUnit->state());
+	
+	return 0;
+}
+
+/** ttsGetSpeechQueue( opt. index )
+ *  Returns the queued text at index.
+ *  If no index exists, returns false.
+ *  If no index provided, returns a table with the entire queue.
+ */
+int TLuaInterpreter::ttsGetSpeechQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+	
+	if(lua_gettop(L) > 0) {
+		if(!lua_isnumber(L, 1)) {
+			lua_pushfstring(L, "ttsGetspeechQueue. bad argument #1 type (optional index as number expected, got %s!)", luaL_typename(L, 1));
+			lua_error(L);
+			return 1;
+		}
+		
+		int index = lua_tonumber(L, 1);
+		index--;
+		
+		if(index < 0 || index > speechQueue.size()) {
+			lua_pushboolean(L, false);
+			return 1;
+		}
+		
+		lua_pushstring(L, speechQueue.at(index).toLatin1().constData());
+		return 1;
+	}
+	
+	lua_newtable(L);
+	
+	for(int i=0;i<speechQueue.size();i++) {
+        lua_pushnumber(L, i + 1);
+        lua_pushstring(L, speechQueue.at(i).toLatin1().constData());
+        lua_settable(L, -3);
+	}
+	
+	return 1;
+}
+
+/** ttsPauseSpeech( opt. pause )
+ *  Pauses speech if passed true, resumes if passed false.
+ *  If no input, returns whether or not the queue is paused.
+ */
+int TLuaInterpreter::ttsPauseSpeech(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+	
+	if(!lua_isboolean(L, 1)) {
+		lua_pushfstring(L, "ttsPauseSpeech: bad argument #1 type (pause expected bool, instead got %s!)", luaL_typename(L, 1));
+		lua_error(L);
+		return 1;
+	}
+	
+	if(lua_toboolean(L, 1))
+		speechUnit->pause();
+	else
+		speechUnit->resume();
+	
+	return 0;
+}
+
+/** ttsClearQueue( opt. index )
+ *  Removes an item from the speech queue at index.
+ *  Returns false if invalid index provided.
+ *	Clears entire queue if 
+ */
+int TLuaInterpreter::ttsClearQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+	
+	if(lua_gettop(L) > 0) {
+		if(!lua_isnumber(L, 1)) {
+			lua_pushfstring(L, "ttsClearQueue: bad argument #1 type (optional index as int expected, got %s!)", luaL_typename(L, 1));
+			lua_error(L);
+			return 1;
+		}
+		
+		int index = lua_tonumber(L, 1);
+		index--;
+		
+		if(index < 0 || index > speechQueue.size()) {
+			lua_pushfstring(L, "ttsClearQueue: index (%d) out of bounds for queue size %d", index+1, speechQueue.size());
```

Comment:
💭 If it is a ___nil+error message___ run-time error case we are beginning to get around to dropping the function identifier (the "ttsClearQueue: " here) from them - but then I'd change "for queue size" to "for text-to-speech queue size" to make the message more comprehensible in a stand-alone context...